test/distributed/org/apache/cassandra/distributed/test/sai/IndexAvailabilityTest.java

```diff
@@ -234,7 +252,324 @@ private void shouldSkipNonQueryableNode(int nodes, List<Integer>... nonQueryable
         }
     }
 
-    private void executeOnAllCoordinatorsAllConsistencies(Cluster cluster, String statement, int liveReplicas, int num) throws Exception
+    @Test
+    public void testAllowFilteringDuringIndexInitialBuildOn3NodeCluster() throws Exception
+    {
+        testAllowFilteringDuringIndexBuildsOn3NodeCluster(true, Index.Status.INITIALIZED);
+    }
+
+    @Test
+    public void testAllowFilteringDuringIndexFullRebuildOn3NodeCluster() throws Exception
+    {
+        testAllowFilteringDuringIndexBuildsOn3NodeCluster(false, Index.Status.FULL_REBUILD_STARTED);
+    }
+
+    public void testAllowFilteringDuringIndexBuildsOn3NodeCluster(boolean isCreateIndex, Index.Status buildStatus) throws Exception
+    {
```

Comment:
This is long and ugly, but covers all cases. I will refactor it soon. It wasn't a priority for now